From cf0e903c898b378365439b410c50dcde74573713 Mon Sep 17 00:00:00 2001
From: Oliver Wolff <oliver.wolff@qt.io>
Date: Wed, 27 Nov 2019 15:03:57 +0100
Subject: [PATCH] dsp/lossless: Fix non gcc ARM builds

The workaround for GCC ARM must not be applied when another toolchain
(like MSVC) is used for the build.

Change-Id: I11ec4558902063ccb085d3f435e24b3a60739dd5
---
 src/dsp/lossless.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/dsp/lossless.c b/src/dsp/lossless.c
index b2fbf17..aad5f43 100644
--- a/src/dsp/lossless.c
+++ b/src/dsp/lossless.c
@@ -81,7 +81,7 @@ static WEBP_INLINE uint32_t ClampedAddSubtractHalf(uint32_t c0, uint32_t c1,
 
 // gcc <= 4.9 on ARM generates incorrect code in Select() when Sub3() is
 // inlined.
-#if defined(__arm__) && LOCAL_GCC_VERSION <= 0x409
+#if defined(__arm__) && defined(__GNUC__) && LOCAL_GCC_VERSION <= 0x409
 # define LOCAL_INLINE __attribute__ ((noinline))
 #else
 # define LOCAL_INLINE WEBP_INLINE
-- 
1.7.1

